service/topology_state_machine.cc

```diff
@@ -261,6 +261,55 @@ future<sstring> topology_state_machine::wait_for_request_completion(db::system_k
     co_return sstring();
 }
 
+void topology_state_machine::generate_cancel_request_update(utils::chunked_vector<canonical_mutation>& muts,
+                                                            gms::feature_service& features,
+                                                            const group0_guard& guard,
+                                                            raft::server_id node,
+                                                            sstring reason) {
+    auto* server_rs = _topology.find(node);
+    if (!server_rs) {
+        return;
+    }
+
+    auto request_id = server_rs->second.request_id;
+    if (!request_id) {
+        return;
+    }
+
+    auto i = _topology.requests.find(node);
+    if (i == _topology.requests.end()) {
+        // Request is not pending, and not cancelable at this stage.
+        return;
```

Comment:
The function also returns here when there is no pending/ongoing request at all, which may be surprising. That's what happens for normal nodes. Here is the reason: https://github.com/scylladb/scylladb/blob/73db5c94dee9945d252fba253f44c90c90a944ae/service/topology_state_machine.hh#L95
Perhaps we can extend this comment to make it clear. It currently looks as if the function returned in the if statement above when there is no pending/ongoing request.